File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.java
class Solution {
    private void dfs(List<List<Integer>> edges, boolean[] seen, int idx, int dist) {
        seen[idx] = true;
        if(seen[dist]) return;

        for(int it : edges.get(idx)) {
            if(!seen[it]) dfs(edges, seen, it, dist);
        }
    }
    
    public boolean validPath(int n, int[][] edges, int source, int destination) {
        boolean[] seen = new boolean[n];
        List<List<Integer>> adj = new ArrayList<>();
        for(int i = 0; i<n; i++) {
            adj.add(new ArrayList<>());
        }

        for(int i = 0; i<edges.length; i++) {
            adj.get(edges[i][0]).add(edges[i][1]);
            adj.get(edges[i][1]).add(edges[i][0]);
        }

        dfs(adj, seen, source, destination);
        return seen[destination];
    }
}

File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.js
/**
 * @param {number} n
 * @param {number[][]} edges
 * @param {number} source
 * @param {number} destination
 * @return {boolean}
 */
var validPath = function(n, edges, source, destination) {
    const seen = new Array(n).fill(false);
    let adj = Array.from({length : n}, () => []);

    for(let it of edges) {
        adj[it[0]].push(it[1]);
        adj[it[1]].push(it[0]);
    }

    dfs = (idx) => {
        seen[idx] = true;
        if(seen[destination]) return;

        for(let edge of adj[idx]) {
            if(!seen[edge]) dfs(edge)
        }
    }

    dfs(source);
    return seen[destination];
};